hs_composite_resource/models.py

```diff
@@ -52,24 +134,299 @@ def get_metadata_xml(self, pretty_print=True, include_format_elements=True):
+        documents need to be created
+        """
+
+        def reset_metadata_is_dirty(aggregation):
+            # resets the metadata.is_dirty to False if the
+            # aggregation is not a NetCDF or Time series aggregation.
+            # NetCDF and Time series aggregations reset metadata.is_dirty
+            # as part of updating the content file
+            if aggregation.get_aggregation_class_name() not in \
+                    ("NetCDFLogicalFile", "TimeSeriesLogicalFile"):
+                aggregation.metadata.is_dirty = False
+                aggregation.metadata.save()
+
+        if aggregation_name is None:
+            for aggregation in self.logical_files:
+                if aggregation.metadata.is_dirty:
+                    aggregation.create_aggregation_xml_documents()
+                    reset_metadata_is_dirty(aggregation)
+        else:
+            try:
+                aggregation = self.get_aggregation_by_name(aggregation_name)
+                if aggregation.metadata.is_dirty:
+                    aggregation.create_aggregation_xml_documents()
+                    reset_metadata_is_dirty(aggregation)
+            except ObjectDoesNotExist:
+                # aggregation_name must be a folder path that doesn't represent an aggregation
+                # there may be single file aggregation in that folder for which xml documents
+                # need to be created
+                self._recreate_xml_docs_for_folder(aggregation_name, check_metadata_dirty=True)
+
+    def _recreate_xml_docs_for_folder(self, folder, check_metadata_dirty=False):
+        """Re-creates xml metadata and map documents associated with the specified folder.
+        If the *folder* represents an aggregation then map and metadata xml documents are
+        recreated only for that aggregation. Otherwise, xml documents are created for any
+        aggregation that may exist in the specified folder and its sub-folders.
+
+        :param  folder: folder for which xml documents need to be re-created
+        :param  check_metadata_dirty: if true, then xml files will be created only if the
+        aggregation metadata is dirty
+        """
+
+        # first check if the the folder represents an aggregation
+        try:
+            aggregation = self.get_aggregation_by_name(folder)
+            if check_metadata_dirty:
+                if aggregation.metadata.is_dirty:
+                    aggregation.create_aggregation_xml_documents()
+            else:
+                aggregation.create_aggregation_xml_documents()
+                # if we found an aggregation by the folder name that means this folder doesn't
+                # have any sub folders as multi-file aggregation folder can't have sub folders
+        except ObjectDoesNotExist:
+            # create xml map and metadata xml documents for all aggregations that exist
+            # in *folder* and its sub-folders
+            if not folder.startswith(self.file_path):
```

Comment:
Should this be done before line 193?

Reply:
No. Line 193 use the relative folder path where as the listing of folder needs the absolute path.